src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs

```diff
@@ -351,7 +358,7 @@ private static bool ContainsUpperCaseLetter(string pattern)
                     }
 
                     var wordSpans = GetWordSpans(candidate);
-                    for (int i = 0; i < wordSpans.Count; i++)
+                    for (int i = 0, n = wordSpans.Count; i < n; i++)
```

Comment:
Yes.  StringBreaks.Count has to actually search the embedded int to find out how many breaks it actaully has.  I think i'd like to make it a method to make it clear that it's not O(1).